client/src/components/share/use-share.tsx

```diff
@@ -0,0 +1,36 @@
+import i18next from 'i18next';
+
+interface ShareTemplateProps {
+  superBlock: string;
+  block: string;
+}
+
+interface ShareTemplateReturnProps {
+  handleRedirectToTwitter: () => void;
+  redirectURL: string;
+}
+
+export const space = '%20';
+export const hastag = '%23';
+export const nextLine = '%0A';
+export const action = 'intent/tweet';
+export const twitterDomain = 'twitter.com';
+export const freecodecampLearnDomainURL = 'www.freecodecamp.org/learn';
+export const twitterDevelpoerDomainURL = 'https://developer.twitter.com';
+
+export const useShare = ({
+  superBlock,
+  block
+}: ShareTemplateProps): ShareTemplateReturnProps => {
+  const redirectFreeCodeCampLearnURL = `https://${freecodecampLearnDomainURL}/${superBlock}/${hastag}${block}`;
+  const i18nSupportedBlock =
+    i18next.t(`intro:${superBlock}.blocks.${block}.title`) || block;
```

Comment:
no need to import everything from `i18next` for `t`, you can use `useTranslation` hook, and there is no need to fallback to `block`, if both of them always true value. 

```suggestion
  const { t } = useTranslation();
  const redirectFreeCodeCampLearnURL = `https://${freecodecampLearnDomainURL}/${superBlock}/${hastag}${block}`;
  const i18nSupportedBlock = t(`intro:${superBlock}.blocks.${block}.title`);

```

Reply:
Note that if both aren't true, then the type should be changed accordingly

Reply:
ok